### DISCUSSION TITLE
MFRONT + SALOME_MECA (aster) - version `GLIBCXX_3.4.26' not found

### DISCUSSION THREAD

**Discussion by Pisolino85** [thelfer/tfel#134] [Q&A]
Dear all,
i'm facing an error while compiling the sovs material model for code_aster and i would like to expose it to you. Here the mfront file:
http://tfel.sourceforge.net/sovs.html

mfront file is compiled with method 1 of : http://tfel.sourceforge.net/aster.html  . from my .comm file:

`path= '/home/argo/SIMULAZIONI/SINTERING/test_case_1/'
os.system('mfront-3.2.1 --obuild '+path+'SkoroholdOlevskyViscousSinteringBehaviour.mfront --interface=aster')
os.system("cp src/libAsterBehaviour.so sovs.so")`

the file sovs.so exists in the tmp/ directory and i'm able to call it back with STAT_NON_LINE.

However an error arises in the log file: 
`/home/argo/salome_meca/V2020.0.1_universal_universal/prerequisites/debianForSalome/lib/libstdc++.so.6: version 'GLIBCXX_3.4.26' not found (required by /tmp/current-code-aster/sovs.so)`

and the error "seems correct" since no 3.4.26 is present : 
`$ strings /home/argo/salome_meca/V2020.0.1_universal_universal/prerequisites/debianForSalome/lib/libstdc++.so.6 | grep GLIBCXX
GLIBCXX_3.4
GLIBCXX_3.4.1
GLIBCXX_3.4.2
GLIBCXX_3.4.3
GLIBCXX_3.4.4
GLIBCXX_3.4.5
GLIBCXX_3.4.6
GLIBCXX_3.4.7
GLIBCXX_3.4.8
GLIBCXX_3.4.9
GLIBCXX_3.4.10
GLIBCXX_3.4.11
GLIBCXX_3.4.12
GLIBCXX_3.4.13
GLIBCXX_3.4.14
GLIBCXX_3.4.15
GLIBCXX_3.4.16
GLIBCXX_3.4.17
GLIBCXX_3.4.18
GLIBCXX_3.4.19
GLIBCXX_3.4.20
GLIBCXX_3.4.21
GLIBCXX_3.4.22
GLIBCXX_DEBUG_MESSAGE_LENGTH`

However why the salome environment call for a library that doesn't have? 

Any idea or solution? 

Thanks in advance.

Andrea

**Comment by thelfer** [comment]
Hi @Pisolino85,

Thanks for your interest to `MFront`. The trouble is that you use your system's compiler when invoking `mfront`. Your compiler will use the system `libstdc++` for linking the `MFront` library. Your system `libstdc++` is later superseeded by the one delivered with `Salome-Méca` when you actually load the library.

I would try the following hack:

~~~~
$ cd /home/argo/salome_meca/V2020.0.1_universal_universal/prerequisites/debianForSalome/lib/
$ mv libstdc++.so.6 libstdc++.so.6_old
$  ln -s /usr/lib/x86_64-linux-gnu/libstdc++.so.6 
~~~~

Could work... or not :)

**Reply by Pisolino85** [reply]
Thank you so much,
indeed you are right, with the substitution now everything works. I'll try to figure out a way to compile it with salome environment also, maybe with a useful automated bash. I would like to bring it to university classroom and i should make it easier

Now the simulation "works": that error went away and now i'm learning how to simulate such sintering behaviour.
I hope to come back with something useful,

Andrea
